python/graphstorm/dataloading/dataset.py

```diff
@@ -64,8 +67,17 @@ def prepare_batch_input(g, input_nodes,
 
         if feat_name is not None:
             # concatenate multiple features together
-            feat[ntype] = th.cat([g.nodes[ntype].data[fname][nid].to(dev) \
-                for fname in feat_name], dim=1)
+            feats = []
+            for fname in feat_name:
+                data = g.nodes[ntype].data[fname]
+                if hasattr(data, 'subcolumn'):  # required by nvshemem
+                    data = data.subcolumn(nid.to(dev)).data
+                elif hasattr(data, 'gather'):
```

Comment:
Please more comments. why if data has attr gather we use data gather?
Can we use isinstance xxx?

Reply:
Added